tajo-core/src/main/java/org/apache/tajo/master/exec/DDLExecutor.java

```diff
@@ -446,11 +447,35 @@ public void alterTable(TajoMaster.MasterContext context, final QueryContext quer
     case SET_PROPERTY:
       catalog.alterTable(CatalogUtil.setProperty(qualifiedName, alterTable.getProperties(), AlterTableType.SET_PROPERTY));
       break;
+    case ADD_PARTITION:
+      existColumnNames(qualifiedName, alterTable.getColumnNames());
+      catalog.alterTable(CatalogUtil.addPartitionAndDropPartition(qualifiedName, alterTable.getColumnNames(),
+        alterTable.getPartitionValues(), alterTable.getLocation(), AlterTableType.ADD_PARTITION));
+      break;
+    case DROP_PARTITION:
+      existColumnNames(qualifiedName, alterTable.getColumnNames());
+      catalog.alterTable(CatalogUtil.addPartitionAndDropPartition(qualifiedName, alterTable.getColumnNames(),
+        alterTable.getPartitionValues(), alterTable.getLocation(), AlterTableType.DROP_PARTITION));
+      break;
     default:
       //TODO
     }
   }
 
+  private boolean existColumnNames(String tableName, String[] columnNames) {
+    for(String columnName : columnNames) {
+      if (!existPartitionColumnName(tableName, columnName)) {
+        throw new NoSuchColumnException(columnName);
+      }
+    }
+    return true;
+  }
+
+  private boolean existPartitionColumnName(String tableName, String columnName) {
+    final TableDesc tableDesc = catalog.getTableDesc(tableName);
+    return tableDesc.getPartitionMethod().getExpressionSchema().contains(columnName) ? true : false;
```

Comment:
The ```?``` conditional operator seems unnecessary.
It's also trivial, but makes an Intellij warning.